### ISSUE TITLE
Generation using velocity prediction and failure.

### ISSUE THREAD

**Issue by theraptorlake** [NVlabs/PixelDiT#12]
I've been trying to implement the novel Pixel DiT architecture using pytorch. But after 100s of failed attempts, I've not been able to make the architecture successfully produce high quality images, I am using L1 loss with velocity prediction which works to some extent, but the images are pretty bad, they're melting, blurry an grayish, I am using the FFHQ dataset with 52K high quality human face images, and I appended the images generated by the model. I am using a batch size of 128, and earlier I used a batch size of 16 but then faced issues, I thought the L1 loss was stalling at 0.1541, so I used gradient accumulation to increase it to 128. I've tried my best to make this architecture work, but I see no light. Could you please help me out here. I tried using REPA too, but it didn't work. The image that you see here is at epoch 63. Does it just need more training, or is it missing something significant? The model is a proper Pixel DiT, with PiT and DiT blocks, for now it's using pure L1 loss and adaptive layer norm instead of RMS norm. Here's what my training script looks like.

Could the authors please help me with this, I really am desperate on this project. I just wanna find out the issue, or the smoking gun. Is it compute? Does it need more epochs? Or is it L2 loss instead of L1.

**Comment by theraptorlake** [comment]
@yeates Could you help me out here please.

**Comment by theraptorlake** [comment]
@wxiong-vision I've spent more than 40-50 days of non stop programming, reading the paper, to implement this architecture, it just never gets to that quality? I've explained above what happened. From my end, I've tried almost everything I could.

**Comment by yeates** [comment]
I don’t see the generated images or training script/config you mentioned, so it’s hard to diagnose the issue from the description alone.

Our training scale was also very different: about 26M images for T2I and 1.28M for ImageNet. I haven’t trained PixelDiT on a 52K-image dataset, so I’m not sure whether the same recipe transfers directly.
